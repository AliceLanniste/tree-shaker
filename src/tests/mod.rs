fn tree_shake(input: &str) {
  let result = crate::tree_shake(input);
  let output = result.codegen_return.source_text;
  insta::assert_snapshot!(output);
}

#[test]
fn test_1() {
  tree_shake(
    r#"
      export let a = 1 && 2;
      export { b };
      let c = 3;
      let { ["b"]: b, d } = { b: 2, d: effect };
      "#,
  );
}
