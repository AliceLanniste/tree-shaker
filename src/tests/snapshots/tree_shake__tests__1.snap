---
source: src/tests/mod.rs
expression: "tree_shake(r#\"\n      export let a = 1 && 2;\n      export { b };\n      let c = 3;\n      let { [\"b\"]: b, d } = { b: 2, d: effect };\n\n      let t = 0;\n      if (t) {\n        effect3;\n      }\n      else {\n        effect4;\n      }\n\n      if (effect) {\n        effect5;\n      }\n\n      while(0) {\n        effect6;\n      }\n\n      while(effect7) {\n        0;\n      }\n      \"#)"
---
export let a = 2;
export { b };
let { b } = {
	b: 2,
	d: effect
};
effect4;
if (effect) effect5;
while (effect7);
